examples/graphics/shader-graph-chunks-builder.html

```diff
@@ -0,0 +1,176 @@
+                    var staticSwitchAlpha = builder.addStaticSwitch(switchName, [ { node: texSampNodeA, swizzle: 'gb' }, { node: texSampNodeB, swizzle: 'ar' } ], switchValue);                
+
+                    // hook up outputs
+                    builder.addOutput( staticSwitchAlpha, 'float', 'sgAlpha' );
+                    builder.addOutput( { node: emisCalcNode, swizzle: 'brgb' }, 'vec3', 'sgEmission');
+
+                    // return built shader graph chunk
+                    var shaderGraphChunk = builder.getShaderGraphChunk();
+
+                    //tmp - directly set private 
+                    // shaderGraphChunk._precision = 'lowp';
+
+                    // replace model's materials with standard node materials
+                    var originalMaterials = [];
+                    var shaderGraphMaterials = [];
+                    
+                    model.meshInstances.forEach(function (meshInstance) {
+                        if (!(meshInstance.material instanceof pc.StandardNodeMaterial)) {
+                            var index = originalMaterials.indexOf(meshInstance.material);
+
+                            if (index === -1) {
+                                index = shaderGraphMaterials.length;
+
+                                originalMaterials.push(meshInstance.material);                            
+                                shaderGraphMaterials.push(new pc.StandardNodeMaterial(meshInstance.material, shaderGraphChunk));
+                                // shaderGraphMaterials.push(new pc.StandardNodeMaterial(meshInstance.material));
+                            }
+
+                            meshInstance.material = shaderGraphMaterials[index];
+                            
+                            meshInstance.material.alphaTest = 0.5;
+
+                            // meshInstance.material.forceFragmentPrecision = "lowp";
+
+                            meshInstance.material.update();
+                        }
+                    });
+
+                    // update time parameter and alpha test every frame
+                    
+                    var flipSwitch = false;
+                    var flipTime = 4.0;
+
+                    app.on("update", function (dt) {
+                        time += dt;
+
+                        if (time>flipTime)
+                        {
+                            flipTime += 4.0;
+                            flipSwitch = true;
+                            switchValue = 1-switchValue;
+                        } 
+
+                        shaderGraphMaterials.forEach(function (material) {
+                            if (flipSwitch)
+                            {
+                                material.overrideGraphSwitch(switchName, switchValue);
+                                material.update();
+                            }
+                            material.overrideGraphParameter('uTime', time);
```

Comment:
I'd prefer these would be set as any other material properties using setParameter to make it simpler for users. A programmer might not know if it is parameter or override, and that can be changed and code would need to be updated.

Reply:
If using the same function name it wouldn't be possible to validate and give warnings if the parameter exists on the graph or not - which could be useful to developers - it also could lead to bugs that occur if the intention wasn't to override a parameter or switch, leading to odd behavior or performance - all of this can also be said of the mesh instance overrides which do use the same function name - so at least it will be consistently tricky to debug :)

Reply:
well a user of a shader does not need to understand internals of the material / graph implementation .. in the future they get a shader graph material from a store, or somebody else on the team builds it, and they just need to set parameters.

Reply:
> small comment, but I'll bring it up just for consideration. These are a few lines of generated shader:
> 
> ```
> vec2 temp_vec2_0;
> vec2 temp_vec2_1;
> temp_vec2_0 = uv0(  );
> temp_vec2_1 = add( float(uTime), vec2(temp_vec2_0) );
> ```
> 
> could this be generated instead? For readability:
> 
> ```
> vec2 temp_vec2_0 = uv0(  );
> vec2 temp_vec2_1 = add( float(uTime), vec2(temp_vec2_0) );
> ```

It could - I would have to add a flag of some sort to indicate the output is a return value - but certainly doable - and make the shader a bit shorter

Reply:
I've made general parameters use setParameters - but I am treating switches differently - because they are different.